src/App.tsx

```diff
@@ -1,24 +1,258 @@
+
+  const handleAddNewTodo = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      if (!todoTitle.trim()) {
+        addError('Title can not be empty');
+
+        return;
+      }
+
+      const newTodo = {
+        userId: USER_ID,
+        title: todoTitle,
+        completed: false,
+      };
+
+      setTempTodos({ ...newTodo, id: 0 });
+      setActiveInput(false);
+
+      addTodo(newTodo)
+        .then(todo => {
+          setAllTodos(todos => [...todos, todo]);
+        })
+        .catch(() => addError('Unable to add a todo'))
+        .finally(() => {
+          setActiveInput(true);
+          setTodoTitle('');
+          setTempTodos(null);
+          loadAllTodos();
+        });
+    }, [todoTitle],
+  );
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            aria-label="clear"
+            type="button"
+            className={classNames(
+              'todoapp__toggle-all',
+              {
+                active: isActiveTodo,
+              },
+            )}
+          />
+
+          <Form
+            handleAddTodo={handleAddNewTodo}
+            todoTitle={todoTitle}
+            setTodoTitle={setTodoTitle}
+            activeInput={activeInput}
+          />
+        </header>
+
+        {(!allTodos.length && isQuery)
+        && (
+          <span style={{
+            display: 'flex',
+            justifyContent: 'center',
+            color: 'grey',
+          }}
+          >
+            You have not any todos
+          </span>
+        )}
+
+        {allTodos.length === 0
+        && errorMessage.length === 0
+        && !isQuery
+          ? (
+            <Loader />
+          ) : (
+            <section className="todoapp__main">
+              <TodoList
+                todos={visibleTodos}
+                tempTodos={tempTodos}
+                handleRemoveTodo={handleRemoveTodo}
+                deletedTodoIds={deletedTodoIds}
+              />
+            </section>
+          )}
+
+        {allTodos.length === 0
+        || (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${countActiveTodos} items left`}
+            </span>
+
+            <nav className="filter">
+              <TodoFilter
+                filteredTodos={filterType}
+                setFilteredTodos={setFilterType}
+              />
+            </nav>
+
+            {isComplitedTodo
```

Comment:
When you remove this button from footer, other footer child elements will shift to the right. You Should make it invisible instead. Look how its done in example

Reply:
Agree